ansible/roles/provision-vm/tasks/redhat.yml

```diff
@@ -1,13 +1,31 @@
 ---
-- name: Set distro
-  set_fact:
-    distro: "centos"
+- name: Get distro info
+  shell: awk -F= '$1=="ID" { print $2 ;}' /etc/os-release
+  register: distro
+
+- name: Get arch info
+  shell: arch
+  register: arch
+
+- name: Register server with Red Hat for IBM Z vms
+  tags: attach_subscription
+  community.general.redhat_subscription:
+    state: present
+    username: "{{ redhat_username }}"
+    password: "{{ redhat_password }}"
+    auto_attach: true
+  register: registration
+  retries: 2
+  delay: 30
+  until: registration is not failed
+  when: arch.stdout == "s390x"
```

Comment:
I think we can use `ansible_architecture` here, instead of finding this information ourselves

Reply:
It might also make sense to add this as a label on the VM during creation so we have this information throughout the various playbooks - I can see that it's used a fair bit and each time we're running `arch` to get it, so would be good to reduce this duplication a bit

Reply:
Good idea, i have set the arch info as group var so it's shared among playbooks/roles/tasks.